test/dialect/AIE/tileop_cse.mlir

```diff
@@ -0,0 +1,26 @@
+//===- tileop_cse.mlir -----------------------------------------*- MLIR -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (C) 2024, Advanced Micro Devices, Inc.
+//
+//===----------------------------------------------------------------------===//
+
+// RUN: aie-opt --split-input-file --pass-pipeline="builtin.module(cse)" %s | FileCheck %s
+
+// CHECK: %[[TILE1:.*]] = aie.tile(1, 1)
+// CHECK-NOT: %[[TILE2:.*]] = aie.tile(1, 1)
+// CHECK: %[[CORE1:.*]] = aie.core(%[[TILE1]])
+// CHECK: %[[CORE2:.*]] = aie.core(%[[TILE1]])
+module {
+  %tile_1 = aie.tile(1, 1)
+  %tile_2 = aie.tile(1, 1)
```

Comment:
It's funny - this should be in a verifier somewhere (prevent duplicate tile definition).

Reply:
Well, don't enable it globally.. I am creating cores with new tiles every time and then I use cse to get rid of the duplicates. Maybe it would be a good check within a `DeviceOp`.

Reply:
I actually like this... will have to think a bit about the consequences.... @jorn would like to see more about how you're using this.

Reply:
@stephenneuendorffer you tagged the wrong person.

I end up with duplicate tiles because I created duplicate cores (from peeling) and then eventually fuse sibling cores on the same tiles.

Reply:
But, then don't you end up with a semantic question of what having 2 aie.core ops in the same tile means?  What meaning are you applying to that, or how are you doing the merging?  Got a repo we can look at?

Reply:
There is still order at the point where I merge them. The meaning is that the core will first execute the operations in the first core block and then the operations in the latter one, so I just merge them into a single core block when they are siblings. https://github.com/nod-ai/iree-amd-aie/blob/c1915c6d54f463fc5bfe4e4612b4246cd90fdff3/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFuseAieRegions.cpp#L98